Jan release 2021 (#1162)
MicrosoftDocs/quantum-docs-pr: articles/resources/relnotes.md

For update instructions, please refer to the [update guide](xref:microsoft.quantum.update).

> ## Version X.XX.XXXXXXXXXX

Reviewer: ```suggestion
## Version 0.15.2101125897
```

*Release date: January 26th, 2021*

- Simplified qubit allocation, providing more convenient syntax for allocating qubits, [see details in Q# language repository](https://github.com/microsoft/qsharp-language/blob/main/Approved/1-implicitly-scoped-qubit-allocation.md).

> - Improved the chemistry library including API improvements and better Python support and integration with existing chemistry packages.

Reviewer 2: This makes it sound like improvements to the Q# API surface, since we've mainly used the term "chemistry library" in that sense (somewhat inaccurately, given the significant C# API surface to that library in particular, but I digress). The improvements made are (@guenp, please correct me if I'm wrong) provided as a new Python library that _does not depend on Q# at all_ for generating problem descriptions.

> - Improved the chemistry library including API improvements and better Python support and integration with existing chemistry packages.

Reviewer 3: ```suggestion
- Created a Python package to work as a convenience layer to the Q# chemistry library, including molecular visualization, a convenience function for parsing Broombridge files for quantum simulation and resource estimation, and functionality to auto-generate input files for several popular chemistry packages, such as NWChem, OpenMolcas and Psi4.
```

> - Q# ergonomic improvements:parentheses on the left-hand side of a curried call are now optional, eliminating the need for parentheses around subsequent calls; parentheses around operation and function type are now optional; parentheses around conditions and in other statement headers are now optional.

Reviewer: ```suggestion
- Q# ergonomic improvements: parentheses on the left-hand side of a curried call are now optional, eliminating the need for parentheses around subsequent calls; parentheses around operation types and function types are now optional; parentheses around conditions and in other statement headers are now optional.
```

> - Q# ergonomic improvements:parentheses on the left-hand side of a curried call are now optional, eliminating the need for parentheses around subsequent calls; parentheses around operation and function type are now optional; parentheses around conditions and in other statement headers are now optional.

Reviewer 4: The optional parentheses for curried calls was in the November release.

For parentheses in statements, it's a little complicated because for statements that take an expression, they're optional (because we can't prevent an expression from being wrapped in parentheses). But for other statements like `for`, `use` and `borrow`, parentheses are actually deprecated and will be removed. Maybe some wording like this would work:

```suggestion
- Parentheses are now optional for operation and function types and `if`, `elif`, `while` and `until` statements. Parentheses for `for`, `use` and `borrow` statements have been deprecated.
```

- Improved width estimates for optimal depth, [see details](https://github.com/MicrosoftDocs/quantum-docs-pr/pull/1159).

> - Released QDK-Python package. 

Reviewer 2: I think this is a new repo, not a new package; @guenp can confirm which packages are now built from that repo. We should also update the contribution guide with those details.

> - Released QDK-Python package. 

Reviewer 3: ```suggestion
- Released QDK-Python package that includes `azure-quantum`, the Python client for submitting quantum-inspired optimization jobs to the Azure Quantum service, as well as `qdk`, including `qdk.chemistry` for visualizing molecules.
```

> 

Reviewer: ```suggestion
- Apply unitary operation provided as explicit matrix using `ApplyUnitary` ([QuantumLibraries#391](https://github.com/microsoft/QuantumLibraries/pull/391), external contribution)

```

Reviewer 2: Can we name the contributor here and link to the API review?

Reviewer 3: @msoeken  would you please create this suggestion again? I think the other commit overwrote the same line. Thanks!

> 

Reviewer 2: ```suggestion
- Fixed https://github.com/microsoft/iqsharp/issues/387 by mitigating performance impact on IQ# kernel startup.
```

> - Fixed https://github.com/microsoft/iqsharp/issues/387 by mitigating performance impact on IQ# kernel startup.

Reviewer: ```suggestion
- Apply unitary operation provided as explicit matrix using `ApplyUnitary` ([QuantumLibraries#391](https://github.com/microsoft/QuantumLibraries/pull/391), external contribution by Dmytro Fedoriaka)
- Fixed https://github.com/microsoft/iqsharp/issues/387 by mitigating performance impact on IQ# kernel startup.
```

For update instructions, please refer to the [update guide](xref:microsoft.quantum.update).

## Version 0.15.2101125897

*Release date: January 26th, 2021*

- Simplified qubit allocation, providing more convenient syntax for allocating qubits, [see details in Q# language repository](https://github.com/microsoft/qsharp-language/blob/main/Approved/1-implicitly-scoped-qubit-allocation.md).

> - Created a Python package to work as a convenience layer to the Q# chemistry library, including molecular visualization, a convenience function for parsing Broombridge files for quantum simulation and resource estimation, and functionality to auto-generate input files for several popular chemistry packages, such as NWChem, OpenMolcas and Psi4.
> - Parentheses are now optional for operation and function types and `if`, `elif`, `while` and `until` statements. Parentheses for `for`, `use` and `borrow` statements have been deprecated.
> - Improved width estimates for optimal depth, [see details](https://github.com/MicrosoftDocs/quantum-docs-pr/pull/1159).
> - Released QDK-Python package that includes `azure-quantum`, the Python client for submitting quantum-inspired optimization jobs to the Azure Quantum service, as well as `qdk`, including `qdk.chemistry`, a Python-based convenience layer for the Q# chemistry library.

Reviewer 3: ```suggestion
- Created QDK-Python repository that includes `azure-quantum`, the Python client for submitting quantum-inspired optimization jobs to the Azure Quantum service, as well as `qdk`, including `qdk.chemistry`, a Python-based convenience layer for the Q# chemistry library that includes molecular visualization and functionality to generate input files for several chemistry packages such as NWChem, Psi4 and OpenMolcas.
- Parentheses are now optional for operation and function types and `if`, `elif`, `while` and `until` statements. Parentheses for `for`, `use` and `borrow` statements have been deprecated.
- Improved width estimates for optimal depth, [see details](https://github.com/MicrosoftDocs/quantum-docs-pr/pull/1159).
```
